pyccel/codegen/printing/ccode.py

```diff
@@ -645,6 +645,54 @@ def _handle_inline_func_call(self, expr):
 
         return code
 
+    def init_stc_container(self, expr, container_var):
+        """
+        Generate the initialization of an STC container in C.
+
+        This method generates and prints the C code for initializing a container using the STC `init()` method.
+
+        Parameters
+        ----------
+        expr : TypedAstNode
+            The object representing the container being printed (e.g., PythonList, PythonSet).
+    
+        container_var : str
+            The variable name to which the container is being assigned.
+        
+        Returns
+        -------
+        str
+            The generated C code for the container initialization.
+        """
+
+        dtype = self.find_in_type_registry(container_var.lhs.class_type)
+        keyraw = '{' + ', '.join(self._print(a) for a in expr.args) + '}'
+        init = f'{container_var.lhs.name} = c_init({dtype}, {keyraw});\n'
```

Comment:
`str` doesn't have a `lhs` method. Are you sure that you describe the correct type here?

Reply:
Ah, my bad. I think at first we were only passing the name attribute to this function, but after we updated it, we were passing the whole variable. The docstring didn't get updated afterward

Reply:
it's actually the whole assignment variable

Reply:
updated 8b4fa11b2c6dce02e30c77cef4d60da0d58e9315